providers/provider_default.go

```diff
@@ -73,8 +73,7 @@ func (p *ProviderData) Redeem(ctx context.Context, redirectURL, code string) (s
 	return
 }
 
-// GetLoginURL with typical oauth parameters
-func (p *ProviderData) GetLoginURL(redirectURI, state string) string {
+func DefaultGetLoginURL(p *ProviderData, redirectURI, state string) (url.URL, url.Values) {
```

Comment:
I think this is a step in the right direction 👍 

Let's refactor it a bit.  Let's move this over to `providers/util.go` with the other provider helpers and call it `makeLoginURL` or something.

It might be over-optimization if it is only used in a couple places, but I'm not a fan of the helper returning the raw `url.URL, url.Values` and requiring the consumers to know to do this:

```
	a, params := DefaultGetLoginURL(p.ProviderData, redirectURI, state)
	// CUSTOMIZE PARAMS HERE
	a.RawQuery = params.Encode()
	return a.String()
```

Maybe make a light struct around this with a few helper methods to do what we need as far as param add/set + `String()`

Reply:
I came up with an alternative form of the method that simply requires to passt `extraParams`, which should make it very clear for the caller what is expected. Pushed 2 commits with the related changes.